presto-main/src/test/java/io/prestosql/server/TestQueryStateInfoResource.java

```diff
@@ -127,12 +142,67 @@ public void testGetQueryStateInfosForUserNoResult()
     public void testGetQueryStateInfo()
     {
         QueryStateInfo info = client.execute(
-                prepareGet().setUri(server.resolve("/v1/queryState/" + queryResults.getId())).build(),
+                prepareGet()
+                        .setUri(server.resolve("/v1/queryState/" + queryResults.getId()))
+                        .setHeader(PRESTO_USER, "unknown")
+                        .build(),
                 createJsonResponseHandler(jsonCodec(QueryStateInfo.class)));
 
         assertNotNull(info);
     }
 
+    @Test
+    public void testGetAllQueryStateInfosDenied()
+    {
+        List<QueryStateInfo> infos = client.execute(
+                prepareGet()
+                        .setUri(server.resolve("/v1/queryState"))
+                        .setHeader(PRESTO_USER, "any-other-user")
+                        .build(),
+                createJsonResponseHandler(listJsonCodec(QueryStateInfo.class)));
+        assertEquals(infos.size(), 2);
+
+        testGetAllQueryStateInfosDenied("user1", 1);
+        testGetAllQueryStateInfosDenied("any-other-user", 0);
+    }
+
+    private void testGetAllQueryStateInfosDenied(String executionUser, int expectedCount)
+    {
+        server.getAccessControl().deny(privilege(executionUser, "query", VIEW_QUERY));
+        try {
+            List<QueryStateInfo> infos = client.execute(
+                    prepareGet()
+                            .setUri(server.resolve("/v1/queryState"))
+                            .setHeader(PRESTO_USER, executionUser)
+                            .build(),
+                    createJsonResponseHandler(listJsonCodec(QueryStateInfo.class)));
+
+            assertEquals(infos.size(), expectedCount);
+        }
+        finally {
+            server.getAccessControl().reset();
+        }
+    }
+
+    @Test
+    public void testGetQueryStateInfoDenied()
+    {
+        server.getAccessControl().deny(privilege("query", VIEW_QUERY));
```

Comment:
when you change the `server` state that suggests that you cannot run tests in paralel. Notice that we used to run such tests within `io.prestosql.tests.AbstractTestQueryFramework#executeExclusively`